helm-chart/binderhub/templates/container-builder/daemonset.yaml

```diff
@@ -0,0 +1,123 @@
+{{- if ne .Values.containerBuilderPod "disabled" -}}
+{{- $builder := (index .Values .Values.containerBuilderPod) }}
+{{- $daemonset := $builder.daemonset }}
```

Comment:
How hard is it to also allow `containerBuilderPod="disabled" and dind.enabled` for backwards compatibility but add a deprecation message? I'd like to get this PR in, and if we can avoid a breaking change that'll make things a bit easier!

Reply:
Less that what I thought would be.

The changes required are not really intrusive and easy to remove once the deprecation period is over.

Reply:
I'm +1 for reduced complexity and a hard failure during template rendering if the removed configuration is observed, where users are provided with information on how to migrate their config.

I wrote about it in https://github.com/jupyterhub/binderhub/pull/1531/files/6794005f7680576b5ac47684822f822d2d711866#r1040093732, its easy and straight forward based on those instructions.

There is no point in deprecating something if we don't actively communicate something to be deprecated etc, and that is not something we will be able to do without changelog and/or warning messages etc.

I think a clear hard failure during template rendering is brilliant, the worst thing that happens is that you are forced to handle a deprecation sooner rather than later when you wish to make an upgrade of binderhub.